src/atomic_integer.rs

```diff
@@ -0,0 +1,86 @@
+use std::sync::atomic::Ordering;
+
+/// Objects that implement this trait must consist exclusively of atomic types
+/// from [`std::sync::atomic`](https://doc.rust-lang.org/std/sync/atomic/), except for
+/// [`AtomicPtr<T>`](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicPtr.html).
```

Comment:
How about AtomicBool?

Reply:
`AtomicBool` is dangerous in this context (i.e. we're essentially casting some memory from the guest to a type), because values other than `0` and `1` for a bool value may lead to undefined behaviour.

Reply:
So it may be better to refine the comments  as "except for [`AtomicPtr<T>`] and AtomicBool<T>".

Reply:
That's totally right. There was no mention of `AtomicBool` initially, but I've added *except for `AtomicPtr` and `AtomicBool` ...* as part of the commit which expands the `AtomicInteger` trait. The message from the commit that removes the `AtomicInteger` for `AtomicBool` impl also offers a bit more details abou the why.